### PR TITLE
CLI: tool to convert PT into TF weights and open hub PR

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

This PR adds a CLI to convert PT weights into TF weights, validate them, and (optionally) open a PR. The open PR part depends on https://github.com/huggingface/huggingface_hub/pull/884, and is wrapped in a try/except for the time being.

Here are 3 PRs open with the tool (`transformers-cli pt-to-tf --model-name [model-name] --local-dir [local-dir] --open-pr`):
1. Text modality: https://huggingface.co/joaogante/test_text/discussions/1
2. Audio modality: https://huggingface.co/joaogante/test_audio/discussions/1
3. Image modality: https://huggingface.co/joaogante/test_img/discussions/1

This tool can also be used to check existing weights. Sadly, there is no programmatic way to check the weights in existing hub PRs, they have to be downloaded manually.

### CODE REVIEW

File: tests/utils/test_cli.py [hunk 1]
```diff
@@ -30,3 +32,22 @@ def test_cli_env(self):
         self.assertIn("Python version", cs.out)
         self.assertIn("Platform", cs.out)
         self.assertIn("Using distributed or parallel set-up in script?", cs.out)
+
+    @is_pt_tf_cross_test
```
Comment (gante) [inline tests/utils/test_cli.py:36]: I'm assuming `tests/utils/test_cli.py` is run with `is_pt_tf_cross_test` at least on `self_push` ([here](https://github.com/huggingface/transformers/blob/main/.github/workflows/self-push.yml))

File: src/transformers/commands/pt_to_tf.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+            type=str,
+            required=True,
+            help="The model name, including owner/organization, as seen on the hub.",
+        )
+        train_parser.add_argument(
+            "--local-dir", type=str, required=True, help="Local directory of the model repository."
+        )
+        train_parser.add_argument(
+            "--open-pr", action="store_true", help="Optional flag to open a PR with converted weights."
+        )
```
Comment (sgugger) [inline src/transformers/commands/pt_to_tf.py:77]: I would make a flag to not do this instead so it's enabled by default?

File: src/transformers/commands/pt_to_tf.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        train_parser.add_argument(
+            "--model-name",
+            type=str,
+            required=True,
+            help="The model name, including owner/organization, as seen on the hub.",
+        )
+        train_parser.add_argument(
+            "--local-dir", type=str, required=True, help="Local directory of the model repository."
+        )
```
Comment (sgugger) [inline src/transformers/commands/pt_to_tf.py:74]: We could have this default to the last part of `model-name`, no?

**Review by sgugger — APPROVED** [review]
Very nice new addition!

File: src/transformers/commands/pt_to_tf.py (hunk 3 above)
Comment (gante) [reply src/transformers/commands/pt_to_tf.py:74]: That would be nice! Like `/tmp/{model-name}`, right?

Comment (sgugger) [reply src/transformers/commands/pt_to_tf.py:74]: For instance, yes!

**Review by LysandreJik — APPROVED** [review]
This looks great! Thanks for working on it @gante!

**Pull request merged**